integr8ly/cloud-resource-operator
# MGDAPI-4545 [Postgres] Delete

## Overview

[MGDAPI-4545](https://issues.redhat.com/browse/MGDAPI-4545)
Adding functionality to delete a postgres instance in GCP.

## Verification
- Provision a GCP cluster
  - From the master branch of the [Delorean](https://github.com/integr8ly/delorean) repo create a file in the root of the repo called gcp_service_account.json. Request GCP credentials and add these to this file.
  - run `make -f make/ocm.mk ocm/cluster.json BYOC=true CLOUD_PROVIDER=gcp`
  - Edit the cluster.json file to update the name (and remove the expiration timestamp if necessary) You can also reduce the number of nodes to 3.
  - Run `make ocm/cluster/create`

- Create a postgres instance (CloudSQL) on GCP.
  - Install [gcloud cli](https://cloud.google.com/sdk/docs/install) 
  - Login and create a cloudsql instance by running `gcloud sql instances create prod-instance --database-version=POSTGRES_9_6 --cpu=2 --memory=8GiB --zone=us-central1-a --root-password=password123` (prod-instance is the instance ID)
  - View the current list of instances `gcloud sql instances list`
  - Alternatively you can view your instance on the GCP [console](https://console.cloud.google.com/sql/instances?project=rhoam-317914).
  
- Delete Postgres Instance
   - In CRO - Checkout this [PR](https://github.com/integr8ly/cloud-resource-operator/pull/530)
  - Log in to your cluster and run `make cluster/prepare`
  - Make sure that the metadata annotations resourceIdentifier in integreatly_v1alpha1_postgres.yaml matches the instance ID from above.
  - Create a postgres cr `PROVIDER=gcp make cluster/seed/postgres`
  - Run CRO `make run`
  - Delete the cr `oc delete postgres prod-instance -n cloud-resource-operator`
  - Check that the gcp postgres instance has been deleted. `gcloud sql instances list`
